DependencyInjection/CompilerPass/PluginPass.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Http\HttplugBundle\DependencyInjection\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Remove plugins if their dependencies are missing.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class PluginPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!($container->hasDefinition('logger') || $container->hasAlias('logger'))) {
```

Comment:
while i agree that theoretically its better to avoid container
parameters, i think the value of clear understandable error messages has
a lot of real value.

we might also find a way to communicate the used plugins directly to the
compiler pass (keep a reference in the bundle class and give it to the
di extension so it can update the compiler pass instance?)